<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SimplifyWindow</class>
 <widget class="QDialog" name="SimplifyWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>686</width>
    <height>536</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Simplify Polygons</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources.qrc">
    <normaloff>:/plugins/CartoGrapher/resources/m.svg</normaloff>:/plugins/CartoGrapher/resources/m.svg</iconset>
  </property>
  <widget class="QWidget" name="horizontalLayoutWidget">
   <property name="geometry">
    <rect>
     <x>-1</x>
     <y>-1</y>
     <width>431</width>
     <height>431</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_5">
    <item>
     <widget class="QTabWidget" name="tabWidget_3">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="Parameters_3">
       <attribute name="title">
        <string>Parameters</string>
       </attribute>
       <widget class="QLabel" name="label_9">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>10</y>
          <width>171</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Input layer or shp</string>
        </property>
       </widget>
       <widget class="QgsMapLayerComboBox" name="mapLayerComboBox_3">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>201</width>
          <height>27</height>
         </rect>
        </property>
       </widget>
       <widget class="QCheckBox" name="select_checkBox_3">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>80</y>
          <width>211</width>
          <height>31</height>
         </rect>
        </property>
        <property name="text">
         <string>Selected features only</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_10">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>120</y>
          <width>171</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Simplify tolerance:</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_11">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>290</y>
          <width>91</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Simplified:</string>
        </property>
       </widget>
       <widget class="QgsFileWidget" name="output_FileWidget_3">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>310</y>
          <width>321</width>
          <height>27</height>
         </rect>
        </property>
       </widget>
       <widget class="QCheckBox" name="output_checkBox_3">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>350</y>
          <width>351</width>
          <height>19</height>
         </rect>
        </property>
        <property name="text">
         <string>Open output file after running algorithm</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
       <widget class="QComboBox" name="simplifyComboBox">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>230</y>
          <width>271</width>
          <height>22</height>
         </rect>
        </property>
        <item>
         <property name="text">
          <string>Douglas-Peucker(Distance)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Visvalingam(Area)</string>
         </property>
        </item>
       </widget>
       <widget class="QLabel" name="label_12">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>200</y>
          <width>191</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Simplification method</string>
        </property>
       </widget>
       <widget class="QDoubleSpinBox" name="tolerance_doubleSpinBox">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>160</y>
          <width>271</width>
          <height>22</height>
         </rect>
        </property>
        <property name="decimals">
         <number>12</number>
        </property>
        <property name="singleStep">
         <double>0.010000000000000</double>
        </property>
        <property name="value">
         <double>1.000000000000000</double>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="Log_3">
       <attribute name="title">
        <string>Log</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="horizontalLayoutWidget_2">
   <property name="geometry">
    <rect>
     <x>430</x>
     <y>0</y>
     <width>251</width>
     <height>431</height>
    </rect>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout_6">
    <item>
     <widget class="QTextEdit" name="textEdit_3">
      <property name="html">
       <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;Simplify  Polygons&lt;/span&gt;&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This algorithm simplifies the geometries in a line or polygon layer.It creates a new layer with less number of vertices.&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Two simplification methods are provied:  Distance based(Douglas-Peucker) and area based(Visvalingam).&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;The tolerance parameter indicates the threshold that based on the map unit to decide whether two nodes will be removed(if distance is smaller than the tolerance) in a segment of the line or polygon.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QPushButton" name="closeBtn">
   <property name="geometry">
    <rect>
     <x>360</x>
     <y>490</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>关闭</string>
   </property>
  </widget>
  <widget class="QProgressBar" name="progressBar">
   <property name="geometry">
    <rect>
     <x>70</x>
     <y>450</y>
     <width>471</width>
     <height>23</height>
    </rect>
   </property>
   <property name="value">
    <number>0</number>
   </property>
   <property name="orientation">
    <enum>Qt::Vertical</enum>
   </property>
   <property name="invertedAppearance">
    <bool>false</bool>
   </property>
   <property name="textDirection">
    <enum>QProgressBar::BottomToTop</enum>
   </property>
  </widget>
  <widget class="QPushButton" name="runBtn">
   <property name="geometry">
    <rect>
     <x>240</x>
     <y>490</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>运行</string>
   </property>
   <property name="default">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPushButton" name="cancelBtn">
   <property name="geometry">
    <rect>
     <x>550</x>
     <y>450</y>
     <width>93</width>
     <height>28</height>
    </rect>
   </property>
   <property name="text">
    <string>Cancel</string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QgsFileWidget</class>
   <extends>QWidget</extends>
   <header>qgsfilewidget.h</header>
  </customwidget>
  <customwidget>
   <class>QgsMapLayerComboBox</class>
   <extends>QComboBox</extends>
   <header>qgsmaplayercombobox.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
</ui>
